Source/WebCore/html/HTMLTemplateElement.cpp

```diff
@@ -98,4 +102,40 @@ void HTMLTemplateElement::didMoveToNewDocument(Document& oldDocument, Document&
     m_content->setTreeScopeRecursively(newDocument.ensureTemplateDocument());
 }
 
+void HTMLTemplateElement::attachAsDeclarativeShadowRootIfNeeded(Element& host)
+{
+    auto modeString = attributeWithoutSynchronization(HTMLNames::shadowrootAttr);
+    std::optional<ShadowRootMode> mode;
+
+    if (equalLettersIgnoringASCIICase(modeString, "closed"_s))
+        mode = ShadowRootMode::Closed;
+    else if (equalLettersIgnoringASCIICase(modeString, "open"_s))
+        mode = ShadowRootMode::Open;
+
+    if (!mode)
+        return;
+
+    bool delegatesFocus = hasAttributeWithoutSynchronization(HTMLNames::shadowrootdelegatesfocusAttr);
+
+    auto exceptionOrShadowRoot = host.attachDeclarativeShadow(*mode, delegatesFocus);
+    if (exceptionOrShadowRoot.hasException())
+        return;
+
+    auto importedContent = document().importNode(content(), /* deep */ true).releaseReturnValue();
+    for (RefPtr<Node> node = NodeTraversal::next(importedContent), next; node; node = next) {
+        next = NodeTraversal::next(*node);
+        if (!is<HTMLTemplateElement>(node))
```

Comment:
`if (!is<HTMLTemplateElement>(*node))` would avoid an unnecessary null check.

Reply:
Good pint. Will do.